pkgs/development/compilers/llvm/7/libcxxabi/default.nix

```diff
@@ -21,6 +22,13 @@ stdenv.mkDerivation {
     export TRIPLE=x86_64-apple-darwin
   '' + lib.optionalString stdenv.hostPlatform.isMusl ''
     patch -p1 -d $(ls -d libcxx-*) -i ${../../libcxx-0001-musl-hacks.patch}
+  '' + lib.optionalString (!stdenv.cc.isClang) ''
+    pushd libcxx-*
+    patch -p2 < ${fetchpatch {
+      url = "https://github.com/llvm/llvm-project/commit/76ccec07b4fa0cc68dfd07d557e7fb661804a468.patch";
+      sha256 = "1lgzkfkp7qinfc6gd8x5di1iq1gqdv81249c6f02chn9q122sbq1";
+    }}
+    popd
```

Comment:
Why is this not done in patches?

Reply:
directory level is likely not correct

Reply:
`stripLen` and `extraPrefix` can do quite a lot in that regard.

Reply:
It seemed more reliable this way, because the extraPrefix would depend on libcxx.version and the FOD hash would become invalid if libcxx.version changed.